jax/_src/lax/slicing.py

```diff
@@ -2058,6 +2092,9 @@ def _dynamic_slice_indices(operand, start_indices: Any):
     if isinstance(i, (int, np.integer)) and core.is_constant_dim(d):
       result.append(lax.convert_element_type(i + d, _dtype(i)) if i < 0 else i)
     else:
-      d = lax.convert_element_type(core.dimension_as_value(d), _dtype(i))
-      result.append(lax.select(i < 0, i + d, i))
+      d = lambda: lax.convert_element_type(core.dimension_as_value(d), _dtype(i))
```

Comment:
Could add comment here saying you're trying to avoid unnecessary staging

Reply:
Done.